compiler/tests-full/stdlib.cma.expected.js

```diff
@@ -5008,15 +5031,16 @@
       /*<<bytes.ml:254:10>>*/ r = [0, x],
       /*<<bytes.ml:255:2>>*/ _P_ =
         /*<<bytes.ml:255:2>>*/ caml_ml_bytes_length(a) - 1 | 0;
+    b:
     if(_P_ >= 0){
      var i = _P_;
      for(;;){
       r[1] =
         /*<<bytes.ml:256:9>>*/ caml_call2
         (f, caml_bytes_unsafe_get(a, i), r[1]);
        /*<<bytes.ml:255:2>>*/  /*<<bytes.ml:255:2>>*/ var _Q_ = i - 1 | 0;
-      if(0 !== i){var i = _Q_; continue;}
-      break;
+      if(0 === i) break b;
```

Comment:
Here, `break` should be enough.

Reply:
I've fixed this case by introducing two "scopes"  when compiling the body of a for loop.
```ocaml
          let scope_stack = (pc, (lab, lab_used, Loop)) :: scope_stack in
          let scope_stack =
            match fall_through with
            | Block fall_through ->
                (fall_through, (lab, lab_used, Exit_loop exit_branch_used)) :: scope_stack
            | Return -> scope_stack
          in
```

Reply:
Yes, this makes a lot of sense.